modules/spacecraftinstruments/util/instrumentdecoder.cpp

```diff
@@ -24,53 +24,43 @@
 
 #include <modules/spacecraftinstruments/util/instrumentdecoder.h>
 
+#include <openspace/documentation/documentation.h>
+#include <openspace/documentation/verifier.h>
 #include <ghoul/misc/assert.h>
 #include <ghoul/logging/logmanager.h>
 #include <ghoul/misc/dictionary.h>
+#include <optional>
 
 namespace {
     constexpr const char* _loggerCat  = "InstrumentDecoder";
-    constexpr const char* KeyDetector = "DetectorType";
-    constexpr const char* KeySpice    = "Spice";
-    constexpr const char* KeyStopCommand = "StopCommand";
+
+    struct [[codegen::Dictionary(InstrumentDecoder)]] Parameters {
+        std::string detectorType;
```

Comment:
Missing documentation? Maybe add a comment? :)

Reply:
I think I prefer to leave it empty for now and have a proper pass over the whole spacecraftinstruments module and actually understand what the properties do and which ones we need to keep